### PR TITLE
Reintroducing encrypted enclave errors

### PR DESCRIPTION
**Pull request by StefanIliev545**
### Why this change is needed

Enclave errors need to be encrypted; This PR reintroduces the initial bits.

### What changes were made as part of this PR

Reintroduced reverted commit; Added propogation of errors all the way to the client. Reworked how the encrypted_client works in order to have a singular way to set results from responses for all clients instead of relying on type switches and reflection.

### PR checks pre-merging

Please indicate below by ticking the checkbox that you have read and performed the required
[PR checks](https://github.com/obscuronet/obscuro-internal/blob/main/dev-ops-docs/dev-pr-checks.md)

- [ ] PR checks reviewed and performed

### CODE REVIEW

File: go/enclave/rpc/rpc_encryption_manager.go [hunk 1]
```diff
@@ -114,3 +115,15 @@ func (rpc *EncryptionManager) AuthenticateSubscriptionRequest(subscription commo
 
 	return nil
 }
+
+func (rpc *EncryptionManager) CreateEncryptorFor(address gethcommon.Address) responses.ViewingKeyEncryptor {
+	encryptorForUser := func(data []byte) ([]byte, error) {
+		result, err := rpc.EncryptWithViewingKey(address, data)
```
Comment (tudor-malene) [inline go/enclave/rpc/rpc_encryption_manager.go:121]: why not just `return rpc.EncryptWithViewingKey(address, data)`

Comment (tudor-malene) [reply go/enclave/rpc/rpc_encryption_manager.go:121]: And then in the main function you can just directly return the lambda.

File: integration/simulation/transaction_injector.go [hunk 2]
```diff
@@ -185,12 +185,17 @@ func (ti *TransactionInjector) issueRandomValueTransfers() {
...
 		ti.stats.Transfer()
 
-		err = obscuroClient.SendTransaction(ti.ctx, signedTx)
+		txHash, err := obscuroClient.SendTransaction(ti.ctx, signedTx)
 		if err != nil {
 			ti.logger.Info("Failed to issue transfer via RPC.", log.ErrKey, err)
 			continue
 		}
 
+		if *txHash != signedTx.Hash() {
```
Comment (tudor-malene) [inline integration/simulation/transaction_injector.go:194]: are you sure this works ? you might need to compare the bytes ( there is a util function)

**Review by tudor-malene — COMMENTED** [review]
some minor comments.
Also, as discussed on chat, the logic in `(c *EncRPCClient) CallContext(` function needs more commenting

**Review by BedrockSquirrel — APPROVED** [review]
LGTM if the e2e tests are passing

**Pull request merged**